dynamic_retriever.py

```diff
@@ -0,0 +1,47 @@
+from langchain.chains.router import MultiRetrievalQAChain
+from langchain.llms import OpenAI
+
+from langchain.embeddings import OpenAIEmbeddings
+from langchain.document_loaders import TextLoader
+from langchain.vectorstores import FAISS
+
+sou_docs = TextLoader('../../state_of_the_union.txt').load_and_split()
+sou_retriever = FAISS.from_documents(sou_docs, OpenAIEmbeddings()).as_retriever()
+
+pg_docs = TextLoader('../../paul_graham_essay.txt').load_and_split()
+pg_retriever = FAISS.from_documents(pg_docs, OpenAIEmbeddings()).as_retriever()
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The file paths provided for the text documents are relative paths. This could lead to issues if the script is run from a different directory. Consider using absolute paths or a method to determine the path dynamically based on the location of the script itself.
```python
import os

base_dir = os.path.dirname(os.path.abspath(__file__))
sou_docs_path = os.path.join(base_dir, '../../state_of_the_union.txt')
pg_docs_path = os.path.join(base_dir, '../../paul_graham_essay.txt')

sou_docs = TextLoader(sou_docs_path).load_and_split()
pg_docs = TextLoader(pg_docs_path).load_and_split()
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->